### PR TITLE
January 2024 refresh

### PR DESCRIPTION
**Pull request by slilichenko**

### CODE REVIEW

File: src/main/java/com/google/solutions/ml/api/vision/BatchRequestsTransform.java [hunk 1]
```diff
@@ -66,7 +67,8 @@ public Integer apply(String input) {
                       return random.nextInt(getKeyRange());
                     }
                   }))
-          .apply("Group Into Batches", GroupIntoBatches.ofSize(getBatchSize()))
+          .apply("Group Into Batches", GroupIntoBatches.<Integer, String>ofSize(getBatchSize())
+              .withMaxBufferingDuration(Duration.standardSeconds(30)))
```
Comment (tjwebb) [inline src/main/java/com/google/solutions/ml/api/vision/BatchRequestsTransform.java:77]: 30sec looks like a magic number -- should it be configurable via pipeline args?

File: src/main/java/com/google/solutions/ml/api/vision/PubSubNotificationToGCSUriDoFn.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public void processElement(ProcessContext c) {
     PubsubMessage message = c.element();
     String eventType = message.getAttribute("eventType");
     if (!Objects.equals(eventType, "OBJECT_FINALIZE")) {
```
Comment (tjwebb) [inline src/main/java/com/google/solutions/ml/api/vision/PubSubNotificationToGCSUriDoFn.java]: here, and on line 60, it'd be nice to have these conditionals in `.filter`s elevated to the main pipeline before this dofn.

I'm also not clear why you're running `.inc()` (ln 51) on a file you may not process.

Comment (tjwebb) [reply src/main/java/com/google/solutions/ml/api/vision/PubSubNotificationToGCSUriDoFn.java]: oh I see, you have a separate rejectedfiles counter. disregard the 2nd part

File: src/main/java/com/google/solutions/ml/api/vision/BatchRequestsTransform.java (hunk 1 above)
Comment (slilichenko) [reply src/main/java/com/google/solutions/ml/api/vision/BatchRequestsTransform.java:77]: Made it configurable.

File: src/main/java/com/google/solutions/ml/api/vision/PubSubNotificationToGCSUriDoFn.java (hunk 2 above)
Comment (slilichenko) [reply src/main/java/com/google/solutions/ml/api/vision/PubSubNotificationToGCSUriDoFn.java]: Addressed the filtering comment.

**Comment by tjwebb** [comment]
lgtm

**Pull request merged**